### PR TITLE
saleeb- homework week3 (added my complete homework js week3)

### PR DESCRIPTION
**Pull request by Saleeb2022**

### CODE REVIEW

File: 1-JavaScript/Week3/homework/ex6-totalCost.js [hunk 1]
```diff
@@ -22,21 +22,39 @@ instead!
...
   // TODO complete this object
+  beer: 1.75,
+  chips: 0.99,
+  cola: 2.4,
+  tea: 3.3,
+  water: 4.6
 };
 
-function calculateTotalPrice(/* TODO parameter(s) go here */) {
+function calculateTotalPrice(price){
```
Comment (cschull) [inline 1-JavaScript/Week3/homework/ex6-totalCost.js:32]: Please make a correction here. You should be passing in cartForParty and deconstructing it in the parameters. A reminder, to deconstruct an object you use this notation {name_of_key, name_of_next_key} for an object that looks like 
{
name_of_key: 1,
name_of_next_key: 2
}

If you have a look at line 56 where you call your method you are not passing in any parameters, and event though you have a price paramater defined in your method you are not using it in your code.

File: 1-JavaScript/Week3/homework/ex7-mindPrivacy.js [hunk 2]
```diff
@@ -30,10 +32,16 @@ const employeeRecords = [
 ];
 
 // ! Function under test
-function filterPrivateData(/* TODO parameter(s) go here */) {
+function filterPrivateData(employeeRecords) {
   // TODO complete this function
+  const nonPrivateRecords = employeeRecords.map((employeeRecords) => {
+   const {gender, salary, ...rest} = employeeRecords;
```
Comment (cschull) [inline 1-JavaScript/Week3/homework/ex7-mindPrivacy.js:38]: this is correct! the only thing I would change it deconstructing like 

{occupation, email, ...rest}
and then return {occupation, email} > your implementation is correct but I would do this to follow what is asked in the instructions! As you can see gender and salary aren't being used so if a variable is not being used, it is best practice not to declare them.

File: 1-JavaScript/Week3/homework/ex2-dogYears.js [hunk 3]
```diff
@@ -16,8 +16,12 @@ calculate it!
...
 -----------------------------------------------------------------------------*/
 
-function calculateDogAge(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+
+function calculateDogAge(age) {
+
+  let dogYears = 7 * age;  
+
+  return ("your doggi is " + dogYears + " years  old in dog years");
```
[the remark this answers is not in the document]
Comment (Saleeb2022) [reply 1-JavaScript/Week3/homework/ex2-dogYears.js:24]: Yes it would have been easier

[the remark this answers is not in the document]
Comment (Saleeb2022) [reply 1-JavaScript/Week3/homework/ex2-dogYears.js:24]: I have fixed  it

File: 1-JavaScript/Week3/homework/ex6-totalCost.js (hunk 1 above)
Comment (Saleeb2022) [reply 1-JavaScript/Week3/homework/ex6-totalCost.js:32]: Thank you, I have fixed it

File: 1-JavaScript/Week3/homework/ex3-tellFortune.js [hunk 4]
```diff
@@ -30,31 +30,42 @@ Note: The DRY principle is put into practice here: instead of repeating the code
...
-function tellFortune(/* TODO add parameter(s) here */) {
+function tellFortune(numKids,partnerName,locations,jobTitles) {
   // TODO complete this function
+  const kids = selectRandomly(numKids);
+  const parents = selectRandomly(partnerName);
+  const location = selectRandomly(locations);
+  const jobTitle = selectRandomly(jobTitles);
+
+  return `You will be a ${jobTitle}, in ${location}, married to ${parents}, with ${kids} kids`;
 }
```
[the remark this answers is not in the document]
Comment (Saleeb2022) [reply 1-JavaScript/Week3/homework/ex3-tellFortune.js:53]: sorry, I do not understand what do you mean here ?

**Review by cschull — APPROVED** [review]

**Review by cschull — APPROVED** [review]

**Pull request closed without merging**